File: backend/Services/Parsers/ColorfleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class ColorfleParser : BasicScoreResultParser
    {
        private ILogger<ColorfleParser> _logger;

        public ColorfleParser(ILogger<ColorfleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Colorfle";
        public override string? HelpText => "Select 'NORMAL' from the difficulty selector.";
        protected override Regex Parser => new Regex($"{GameName} \\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => null;
        public override string Url => "https://colorfle.com/";
    }
}

File: lambdas/src/parsers/colorfle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class ColorfleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Colorfle';
  readonly helpText = "Select 'NORMAL' from the difficulty selector.";
  protected readonly extraContent = null;
  readonly url = 'https://colorfle.com/';
  
  // Matches "Colorfle 773 X/6" or "Colorfle 624 2/6"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} \\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
